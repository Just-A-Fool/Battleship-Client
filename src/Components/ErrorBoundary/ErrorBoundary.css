.errorHandler {
  margin-top: 200px;
}