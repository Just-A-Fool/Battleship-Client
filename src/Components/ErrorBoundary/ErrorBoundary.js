import React from 'react';
import Header from '../Header/Header';
import { Link } from 'react-router-dom';
import './ErrorBoundary.css';

class ErrorBoundary extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      hasError: false
    };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {
    if(this.state.hasError){
      return (
        <div>
        <Header />
        <h3 className='errorHandler'> Uh oh! We've had a misfire! Please try returning to
          the homepage...
        </h3>
        <button>
          <Link to='/'>Home</Link>
        </button>
        </div>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;