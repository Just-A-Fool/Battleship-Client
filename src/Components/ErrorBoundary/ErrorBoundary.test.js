import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ErrorBoundary from './ErrorBoundary.js'

describe(`Error Boundary Component`, () => {
    it(`Renders without errors`, () => {
        const wrapper = shallow(<ErrorBoundary />)
        wrapper.setState({hasError:true})
        expect(wrapper.text()).toMatchSnapshot()
    })

    it(`Displays the error boundary component when rendered`, () => {
        const wrapper = shallow(<ErrorBoundary />)
        wrapper.setState({hasError:true})
        setTimeout(() =>{
            expect(toJson(wrapper)).toMatchSnapshot()
        },500)
    })
    
    
    it(`Displays 'Uh oh! We've had a misfire! Please try returning to the homepage...' message on error`, () => {
        const wrapper = shallow(<ErrorBoundary />)
        wrapper.setState({hasError:true})
        expect(wrapper.text()).toMatch(/We've had a misfire!/)
    })

});