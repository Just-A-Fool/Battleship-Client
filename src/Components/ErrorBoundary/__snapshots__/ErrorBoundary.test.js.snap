// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary Component Renders without errors 1`] = `"<withRouter(Header) /> Uh oh! We've had a misfire! Please try returning to the homepage...<Link />"`;
